### DISCUSSION TITLE
Implications of unstable file format

### DISCUSSION THREAD

**Discussion by ghost** [flxzt/rnote#870] [Q&A]
Am I understanding correctly: if file format ever changes again, I will not be able to use it in new versions of Rnote *and* there's no way to migrate/upgrade file from previous to new format?

**Comment by flxzt** [comment]
There will likely be a break for version `v1.0` to replace the current ad-hoc implementation. But otherwise we are working as hard as possible to always provide an upgrade path on minor version updates. But yes, I wanted to leave me the possibility to break at some point for a more stable, spec'd file format.

**Reply by GGORG0** [reply]
Will there be a migration tool or something?

**Reply by flxzt** [reply]
That's the plan, but I can't guarantee it
